src/Maestro/SubscriptionActorService/PullRequestActor.cs

```diff
@@ -867,32 +869,35 @@ private async Task<string> CreatePullRequestAsync(List<UpdateAssetsParameters> u
             // non-coherency updates is 1 then combine coherency updates with those.
             // Otherwise, put all coherency updates in a separate commit.
             bool combineCoherencyWithNonCoherency = (nonCoherencyUpdates.Count == 1);
+            int startingReferenceId = 1;
+          
             foreach ((UpdateAssetsParameters update, List<DependencyUpdate> deps) in nonCoherencyUpdates)
             {
                 var message = new StringBuilder();
                 List<DependencyUpdate> dependenciesToCommit = deps;
                 await CalculateCommitMessage(update, deps, message);
-
+                Build build = await GetBuildAsync(update.BuildId);
 
                 if (combineCoherencyWithNonCoherency && coherencyUpdate.update != null)
                 {
                     await CalculateCommitMessage(coherencyUpdate.update, coherencyUpdate.deps, message);
-                    await CalculatePRDescription(coherencyUpdate.update, coherencyUpdate.deps, null, description);
+                    startingReferenceId += PullRequestDescriptionBuilder.CalculatePRDescription(coherencyUpdate.update, coherencyUpdate.deps, null, description, build, startingReferenceId);
```

Comment:
nit: You could also turn `startingReferenceId` into a `ref` argument and make sure it gets incremented properly as part of `CalculatePRDescription` so that it's more clear you need to pass this along between the calls

Reply:
I moved it inside the class

Reply:
Even better